core/src/main/java/org/apache/accumulo/core/file/DispatchingFileFactory.java

```diff
@@ -60,78 +54,54 @@ private FileOperations findFileFactory(String file) {
     }
   }
 
-  @Override
-  public FileSKVIterator openIndex(String file, FileSystem fs, Configuration conf, AccumuloConfiguration acuconf) throws IOException {
-    return findFileFactory(file).openIndex(file, fs, conf, acuconf, null, null);
-  }
-
-  @Override
-  public FileSKVIterator openReader(String file, boolean seekToBeginning, FileSystem fs, Configuration conf, RateLimiter readLimiter,
-      AccumuloConfiguration acuconf) throws IOException {
-    FileSKVIterator iter = findFileFactory(file).openReader(file, seekToBeginning, fs, conf, readLimiter, acuconf, null, null);
-    if (acuconf.getBoolean(Property.TABLE_BLOOM_ENABLED)) {
-      return new BloomFilterLayer.Reader(iter, acuconf);
+  /** If the table configuration disallows caching, rewrite the options object to not pass the caches. */
+  private static <T extends FileReaderOperation<T>> T filterCaches(T input) {
+    BlockCache indexCache = input.getIndexCache();
+    BlockCache dataCache = input.getDataCache();
+    if (!input.getTableConfiguration().getBoolean(Property.TABLE_INDEXCACHE_ENABLED)) {
+      indexCache = null;
```

Comment:
Nevemind.  I thought this had no impact, but I was looking at the subtracted lines of the diff.  Now I see that it does have an impact.